PyTorch build requires Numpy now

When I don't have a working numpy dependency detected (due to unrelated shenanigans), the build fails with this:

```
-- Building version 0.2.0+803afd5
running egg_info
writing torch.egg-info/PKG-INFO
writing dependency_links to torch.egg-info/dependency_links.txt
writing requirements to torch.egg-info/requires.txt
writing top-level names to torch.egg-info/top_level.txt
reading manifest file 'torch.egg-info/SOURCES.txt'
writing manifest file 'torch.egg-info/SOURCES.txt'
running build_ext
-- NumPy not found

....


/private/home/ezyang/pytorch/torch/csrc/generic/Tensor.cpp: In function ‘THCudaTensor* THCPFloatTensor_fromNumpy(PyObject*)’:
/private/home/ezyang/pytorch/torch/csrc/generic/Tensor.cpp:159:3: error: ‘PyArrayObject’ was not declared in this scope
   PyArrayObject *array = (PyArrayObject*)numpy_array;
   ^
```

And then a big pile of related errors.

CC @colesbury @gchanan

--- a/torch/csrc/generic/Tensor.cpp
+++ b/torch/csrc/generic/Tensor.cpp
@@ -147,7 +147,7 @@ static void THPTensor_(setInconsistentDepthError)(std::vector<size_t> &sizes,
   THPUtils_setError(error.c_str());
 }
 
-#if defined(NUMPY_TYPE_ENUM) || defined(THC_GENERIC_FILE)
+#ifdef WITH_NUMPY
 
 #ifndef THC_REAL_IS_HALF
 #define load_real real

--- a/torch/csrc/generic/Tensor.cpp
+++ b/torch/csrc/generic/Tensor.cpp
@@ -1,4 +1,5 @@
 #ifndef TH_GENERIC_FILE
+
 #define TH_GENERIC_FILE "generic/Tensor.cpp"
 #else
 
@@ -147,7 +148,7 @@ static void THPTensor_(setInconsistentDepthError)(std::vector<size_t> &sizes,
   THPUtils_setError(error.c_str());
 }
 
-#ifdef WITH_NUMPY
+#if defined(NUMPY_TYPE_ENUM) || (defined(WITH_NUMPY) && defined(THC_GENERIC_FILE))
 
 #ifndef THC_REAL_IS_HALF
 #define load_real real

